crawler/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sync"
+
+	"golang.org/x/net/html"
+)
+
+type Crawler struct {
+	visited      map[string]bool
+	brokenLinks  int
+	brokenReport []string
+	mu           sync.Mutex
+}
+
+func NewCrawler() *Crawler {
+	return &Crawler{
+		visited:      make(map[string]bool),
+		brokenReport: []string{},
+	}
+}
+
+func (c *Crawler) Visit(link string) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.visited[link] {
+		return false
+	}
+	c.visited[link] = true
+	return true
+}
```

Comment:
Feels like this logic should be the other way around. If we have already visited the link, return true otherwise false?
```suggestion
	c.mu.Lock()
	defer c.mu.Unlock()
	if c.visited[link] {
		return true
	}
	c.visited[link] = false
	return false
}
```

Reply:
Visit returns true if it visits the link, False if not (already visited)